### PR TITLE
deprecated ospaths

### PR DESCRIPTION
**Pull request by Araq**

### CODE REVIEW

**Comment by dom96** [comment]
Why are you creating these pointless PRs? They are just spamming my notifications :/

**Comment by timotheecour** [comment]
/cc @Araq 
actually merging commits through PR (like everyone else does) is better than directly committing as it's easier for referencing (eg #9665), commenting etc. To reduce spamming, you could just submit them from your own private fork instead of through a branch; also it's not so great to have private branches (araq-deprecated-ospaths) exposed on this public repo 

Note: if for some reason https://github.com/araq/Nim is just an alias for  https://github.com/nim-lang/Nim you could also create an organization and push from there, eg see snapshot below where I create a fork of something already in my account

image

File: lib/pure/os.nim [hunk 1]
```diff
@@ -17,33 +17,701 @@
...
 
   proc toTime(ts: Timespec): times.Time {.inline.} =
     result = initTime(ts.tv_sec.int64, ts.tv_nsec.int)
-
 else:
   {.error: "OS module not ported to your operating system!".}
 
-import ospaths
-export ospaths
+when defined(nimscript) and defined(nimErrorProcCanHaveBody):
```
Comment (timotheecour) [inline lib/pure/os.nim]: /cc @Araq @dom96 
should that be:
```nim
when defined(nimscript):
  when defined(nimErrorProcCanHaveBody):
    {.pragma: noNimScript, error: "this proc is not available on the NimScript target".}
  else:
    {.pragma: noNimScript, error.}
else:
  {.pragma: noNimScript.}
```

Comment (Araq) [reply lib/pure/os.nim]: No, ``nimErrorProcCanHaveBody`` means the Nim compiler allows .error procs to have a body that is then not sem'checked. Very useful here.

**Pull request merged**